ml4h/models/merge_blocks.py

```diff
@@ -153,6 +159,22 @@ def pairwise_cosine_difference(t1, t2):
     return K.mean(tf.acos(dot))
 
 
+def contrastive_difference(left: Tensor, right: Tensor, batch_size: int, temperature: Tensor):
```

Comment:
am i remembering right that you were getting some numerical issues in this loss? Maybe the trick of subtracting a large constant before applying the softmax could help

Reply:
cool Idk that trick :)